explorer/app/components/AzulFileDownload/azulFileDownload.tsx

```diff
@@ -19,7 +19,8 @@ export const AzulFileDownload = ({
   url,
 }: AzulFileDownloadProps): JSX.Element => {
   const downloadRef = useRef<HTMLAnchorElement>(null);
-
+  // Used to prevent the download button from being clicked twice
+  const [isDownloading, setDownloading] = useState(false);
```

Comment:
This is great work @jpaten! The only small change I would recommend is changing `setDownloading` to `setIsDownloading` so that the naming convention is exactly the same as the state `isDownloading`. See [here](https://beta.reactjs.org/learn/state-as-a-snapshot#setting-state-triggers-renders) as an example.